### PR TITLE
[9.0] [MIG] web_favicon

### PR DESCRIPTION
**Pull request by pedrobaeza**
This includes the full 8.0 module + the little migration needed, that is on the commit https://github.com/OCA/web/commit/28dbb5c80cbf6484ef6bfda9e9502eb0fa93325e

### CODE REVIEW

**Comment by hbrunn** [comment]
:+1:

**Comment by legalsylvain** [comment]
Thanks a lot for the cherry pick and the unique commit to review.
:+1: 
Just an newbie question on icons. I don't understand why it's necessary to set so much icons. One is not sufficiant ? specially for a so simple icon ! 
https://github.com/OCA/web/pull/344/files#diff-4b6d76cb48d837f5cc600cfa84d70033R4

**Comment by pedrobaeza** [comment]
@legalsylvain, for now that icons are not active, but the purpose is that each platform uses the optimized icon for it (that it's to say, the exact resolution).

**Comment by adrienpeiffer** [comment]
:+1: (Code review and functional test)

File: web_favicon/static/src/img/website_favicon_sample.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <link rel="shortcut icon" href="/website_favicon/static/src/img/favicon.ico" type="image/x-icon"/>
+    <link rel="apple-touch-icon" sizes="57x57" href="/website_favicon/static/src/img/apple-touch-icon-57x57.png"/>
+    <link rel="apple-touch-icon" sizes="60x60" href="/website_favicon/static/src/img/apple-touch-icon-60x60.png"/>
+    <link rel="apple-touch-icon" sizes="72x72" href="/website_favicon/static/src/img/apple-touch-icon-72x72.png"/>
+    <link rel="apple-touch-icon" sizes="76x76" href="/website_favicon/static/src/img/apple-touch-icon-76x76.png"/>
+    <link rel="apple-touch-icon" sizes="114x114" href="/website_favicon/static/src/img/apple-touch-icon-114x114.png"/>
+    <link rel="apple-touch-icon" sizes="120x120" href="/website_favicon/static/src/img/apple-touch-icon-120x120.png"/>
+    <link rel="apple-touch-icon" sizes="144x144" href="/website_favicon/static/src/img/apple-touch-icon-144x144.png"/>
+    <link rel="apple-touch-icon" sizes="152x152" href="/website_favicon/static/src/img/apple-touch-icon-152x152.png"/>
+    <link rel="apple-touch-icon" sizes="180x180" href="/website_favicon/static/src/img/apple-touch-icon-180x180.png"/>
```
Comment (lasley) [inline web_favicon/static/src/img/website_favicon_sample.html:13]: We may want to use `apple-touch-icon-precomposed` here instead - it prevents iOS from adding default effects, which provides more control over the display of your icon.

File: web_favicon/static/src/img/website_favicon_sample.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <link rel="icon" type="image/png" href="/website_favicon/static/src/img/favicon-32x32.png" sizes="32x32"/>
+    <link rel="icon" type="image/png" href="/website_favicon/static/src/img/android-chrome-192x192.png" sizes="192x192"/>
+    <link rel="icon" type="image/png" href="/website_favicon/static/src/img/favicon-96x96.png" sizes="96x96"/>
+    <link rel="icon" type="image/png" href="/website_favicon/static/src/img/favicon-16x16.png" sizes="16x16"/>
+    <link rel="manifest" href="/manifest.json"/>
+    <meta name="msapplication-TileColor" content="#da532c"/>
+    <meta name="msapplication-TileImage" content="/mstile-144x144.png"/>
+    <meta name="theme-color" content="#ffffff"/>
```
Comment (lasley) [inline web_favicon/static/src/img/website_favicon_sample.html:21]: Apple meta?

```
<meta content="yes" name="apple-mobile-web-app-capable">
<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
```

File: web_favicon/static/src/img/website_favicon_sample.html (hunk 1 above)
Comment (pedrobaeza) [inline web_favicon/static/src/img/website_favicon_sample.html:13]: This is not in use, so you can change it as you want. This is only a suggested template extracted from a website.

File: web_favicon/static/src/img/website_favicon_sample.html (hunk 2 above)
Comment (pedrobaeza) [inline web_favicon/static/src/img/website_favicon_sample.html:21]: The same.

File: web_favicon/static/src/img/website_favicon_sample.html (hunk 1 above)
Comment (lasley) [inline web_favicon/static/src/img/website_favicon_sample.html:13]: Ohhhh I guess looking at the file name would have helped. My bad

**Comment by lasley** [comment]
👍  - functional and code

@pedrobaeza Mobile ready is a blocker for me, so I'll pick up on this in terms of implementing that. Was any effort put forth towards that aside from the static templates?

**Comment by pedrobaeza** [comment]
Not for now, so you're welcome to add it in a new PR starting from this, or making a PR over my branch.

**Comment by lasley** [comment]
Cool yup that's what I was planning, just making sure that you didn't hit some sort of roadblock or something that I needed to know about before starting.

@hbrunn mentioned that this should be an option, I am assuming the community agrees? I'm struggling to understand the use case behind intentionally not supporting all devices.

**Comment by pedrobaeza** [comment]
Yeah, of course. @hbrunn only refers that the minimum required would be current module, but if you add some extra features (as optional), they're always welcome.

**Comment by lasley** [comment]
Got it, thanks @pedrobaeza

**Comment by rafaelbn** [comment]
I think this PR is ready to merge, isn't it?

**Pull request merged**